src/main/webapp/app/index.tsx

```diff
@@ -74,48 +75,83 @@ superagent.Request.prototype.end = function (callback) {
       response.body = response.text;
     }
 
-    // If the code is 401, which means the token has expired, we need to refresh the page
-    // But in certain pages, 401 is a valid response
+    // Reload is not neccessary in dev environment
+    if (AppConfig.serverConfig.appProfile !== AppProfile.PROD) {
+      return callback(error, response);
+    }
+
+    // Only reload if the status code is 401, indicating the token might have expired.
+    if (response?.statusCode !== 401) {
+      return callback(error, response);
+    }
+
+    // We do not reload the page when the /api/authenticate fails
+    // or on auth-related pages where 401 is a valid response
     if (
-      response &&
-      response.statusCode === 401 &&
-      AppConfig.serverConfig.token &&
-      AppConfig.serverConfig.appProfile === AppProfile.PROD &&
-      response.req &&
-      // we do not reload the page when the /api/authenticate failed
-      response.req.url !== API_ROUTE.AUTHENTICATE &&
+      response?.req?.url === API_ROUTE.AUTHENTICATE ||
       UNAUTHORIZED_ALLOWED_PATH.some(path =>
         window.location.pathname.endsWith(path)
       )
     ) {
```

Comment:
```
UNAUTHORIZED_ALLOWED_PATH.some(path =>
        window.location.pathname.endsWith(path)
      )
```

The problem was that the UNAUTHORIZED_ALLOWED_PATH check was inverted relative to what we actually want. UNAUTHORIZED_ALLOWED_PATH is a list of routes where 401 is expected and should not trigger a full page reload (login, password reset, verification, register).

In other words, the old logic was saying "refresh the page if we're on the login, reset, or register pages, but don't refresh if we're on the gene page, home page, etc"